src/Engine/Common/IRenderContextImp.cs

```diff
@@ -442,6 +454,22 @@ public interface IRenderContextImp
         /// <exception cref="System.ArgumentException">Vertices must not be null or empty</exception>
         void SetVertices(IMeshImp mesh, float3[] vertices);
 
+        /// <summary>
+        /// Creates and binds the instance model matrices onto the GL render context and assigns an buffer object index to the passed <see cref="IInstanceDataImp" /> instance.
+        /// </summary>
+        /// <param name="instanceImp">The <see cref="IInstanceDataImp"/> instance.</param>
+        /// <param name="instancePositions">The instance positions.</param>
+        /// <param name="instanceRotations">The instance rotations.</param>
+        /// <param name="instanceScales">The instance scale values.</param>
+        public void SetInstanceTransform(IInstanceDataImp instanceImp, float3[] instancePositions, float3[] instanceRotations, float3[] instanceScales);
```

Comment:
Why no `float4x4[] instanceTransform`?

Reply:
Maybe when we make the InstanceData arrays observable :)